### ISSUE TITLE
MA0016 warns also for protected properties and methods

### ISSUE THREAD

**Issue by Piedone** [meziantou/Meziantou.Analyzer#280]
**Context**
Rule identifier: _MA0016_
Target Framework: _.NET 4.6.1_
Language version: _7.3_

**Describe the bug**
The [MA0016 docs](https://github.com/meziantou/Meziantou.Analyzer/blob/main/docs/Rules/MA0016.md) says:

> **Public** methods that returns a collection implementation, such as List HashSet or Dictionary<TKey,TValue>, or have a parameter of one of these types.

However, it also warns for protected properties and methods.

**To Reproduce**

````c#
    public class Test
    {
        protected List<string> MyProperty { get; set; }

        protected void MyMethod(List<string> list)
        {
            MyProperty = list;
        }
    }
````

Both `MyProperty` and `MyMethod` causes a violation. Note that if I change them to `private` they correctly don't cause a violation.

**Additional context**
N/A

**Comment by meziantou** [comment]
A `protected` method is accessible to other assemblies (in derived classes), so it's kind of public. Note that MA0016 should not report for `private protected` methods.

The documentation should be more precise about what public means.

**Comment by Piedone** [comment]
Ah I see. Then indeed, just the docs need to be updated. I suggest the wording "non-private".

**Comment by Piedone** [comment]
Thank you!
